### PR TITLE
Swift3 Update

### PR DESCRIPTION
**Pull request by kevindelord**
- [x] Migrate sample code to Swift 3.
- [x] Add new get_prefix_function Swiftlint rules.
- [x] Make `guard` more important than `if let`.
- [x] Function naming in Swift 3.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -221,54 +329,118 @@ class app_widgetContainer {
...
+### Nested structs
+
+When declaring a `struct` or `enum` within a scope **only the top level "owner" is UpperCamelCase**.
+
+The inner structs use **lowerCamelCase**.
+
+```swift
+struct Database { // UpperCamelCase
+
+    struct key { // lowerCamelCase
```
Comment (tschob) [inline README.markdown]: Why should inner structs or Enums be lowerCamelCase? They are still types and should therefore be upperCamelCase (https://swift.org/documentation/api-design-guidelines/ > Follow case conventions)

File: README.markdown [hunk 2]
```diff
@@ -459,26 +648,43 @@ For more information about the documenting your code on Swift, please read this
 
 ### Mark
 
-The mark should be use as much as possible to actually separate and structure your classes and code within different files.
+The mark should be use as much as possible to actually separate and structure your classes and related code extensions within different files.
```
Comment (tschob) [inline README.markdown]: typo `use` -> `used`

File: README.markdown [hunk 3]
```diff
@@ -1020,401 +1229,334 @@ list.sort { (a, b) -> Bool in
...
 ```swift
-if (indexes.count >= 0),
-    let _users  = users where (_users.count > 0),
-    let _max    = max where (_max > _users.count) {
-        print(_users)
+// Optional Binding statement
+if (indexes.isEmpty == false),
+    let _users = users, (_users.isEmpty == false),
+    let _max = max, (_max > _users.count) {
+    print(_users)
```
Comment (tschob) [inline README.markdown]: indention

**Review by tschob — CHANGES_REQUESTED** [review]

File: README.markdown (hunk 1 above)
Comment (kevindelord) [reply README.markdown]: Well isn't how the native API is made of? `thisClass.thisKindOfState.thisValue.thisFunction` ?

Comment (kevindelord) [reply README.markdown]: My bad, I receive a Swiftlint warning asking for the inner struct to be UpperCase.
Good job !

**Pull request merged**